x-pack/platform/plugins/shared/fleet/common/experimental_features.ts

```diff
@@ -23,38 +21,73 @@ const _allowedExperimentalValues = {
  */
 export const allowedExperimentalValues = Object.freeze<
   Record<keyof typeof _allowedExperimentalValues, boolean>
 >({ ..._allowedExperimentalValues });
 
+export type ExperimentalFeatures = typeof allowedExperimentalValues;
+
 type ExperimentalConfigKey = keyof ExperimentalFeatures;
 type ExperimentalConfigKeys = ExperimentalConfigKey[];
-type Mutable<T> = { -readonly [P in keyof T]: T[P] };
 
 const allowedKeys = Object.keys(allowedExperimentalValues) as Readonly<ExperimentalConfigKeys>;
 
 /**
- * Parses the string value used in `xpack.fleet.enableExperimental` kibana configuration,
- * which should be a string of values delimited by a comma (`,`)
+ * Parses two settings in kibana configuration:
+ * 1. xpack.fleet.enableExperimental: an array of experimental values to enable
+ * 2. xpack.fleet.experimentalFeatures: an object that associates a boolean to experimental values to enable or disable them
  *
- * @param configValue
+ * The objective is to make xpack.fleet.experimentalFeatures the source of truth, while keeping
+ * xpack.fleet.enableExperimental for backward compatibility.
+ * In case of conflict, xpack.fleet.experimentalFeatures takes precedence over xpack.fleet.enableExperimental.
+ *
+ * @param enableExperimentalConfigValue the value of xpack.fleet.enableExperimental
+ * @param experimentalFeaturesConfigValue the value of xpack.fleet.experimentalFeatures
+ * @returns an object with experimental features and their boolean values
  */
-export const parseExperimentalConfigValue = (configValue: string[]): ExperimentalFeatures => {
-  const enabledFeatures: Mutable<ExperimentalFeatures> = { ...allowedExperimentalValues };
+export const parseExperimentalConfigValue = (
+  enableExperimentalConfigValue: string[],
+  experimentalFeaturesConfigValue: Partial<ExperimentalFeatures>
+): ExperimentalFeatures => {
+  return getUpdatedExperimentalFeatures(
+    enableExperimentalConfigValue,
+    experimentalFeaturesConfigValue,
+    allowedExperimentalValues
+  );
+};
+
+/**
+ * Parses and merges two config values into a single object representing experimental features.
+ * Made to be generic for easier testing.
+ *
+ * @param enableExperimentalConfigValue an array of strings that are experimental feature values
+ * @param experimentalFeaturesConfigValue an object with string keys (experimental feature names) and boolean values
+ * @param existingFeatures an object with pre-existing experimental features and their default values
+ * @returns an object with experimental features values updated based on the two config values provided
+ */
+export const getUpdatedExperimentalFeatures = <T extends Record<string, boolean>>(
+  enableExperimentalConfigValue: string[],
+  experimentalFeaturesConfigValue: { [k: string]: boolean },
+  existingFeatures: T
+): T => {
+  const updatedFeatures = { ...existingFeatures };
+
+  for (const value of enableExperimentalConfigValue) {
+    if (value in existingFeatures) {
```

Comment:
nit: I think using hasOwnProperty is a little safer as it do not check for inherited properties 
```
        Object.prototype.hasOwnProperty.call(existingFeatures, value)
```